Android client. Fixed and recoded the channel prop dialog

amirmahdifard opened the pull request:
this fixes https://github.com/BearWare/TeamTalk5/issues/800
fixes https://github.com/BearWare/TeamTalk5/discussions/2502
Channel prop has been recoded, to make many advantages, fixes, new options, and many improovments! The following has happened
1. Added fixed audio volume for all users option support as well as the slider when enabled to controle the fixed volume exactly like desktop clients, enjoy who has requested and wanted this feature a lot! @kirill-jjj @BlindMaster24 I remember of you guys for now to mention!
2. Ability to set voice and media file max duration timeout timer which is only supported in qt teamtalk, it's a new option of teamtalk that has been added recently!
3. People who had any kind of problems with not being able to creat stereo channels, and any kind of other controles that was buggy due to 2 difrent dialogs was trying to comunicate with each other, I Currently remember of these people who had problem. @CoBC @jeremyp3 in the issue i've linked this pull request to fix it.
4. Now, channel prop dialog looks exactly like the channel prop of the desktop clients, to keep the sink between clients and give them same functionality. This also has some other advantages too. when you set an item, you can really easyly correct it, no need to press a button to go to another dialog.
5. Code cleanup and easy controle, audio codec activity along with its 5 following xml files has been removed.
Guys, please try this new dialog and see if your problems has been solved by this new dialog completely? Thanks!

amirmahdifard commented:
oops. Let me see what was going wrong there

amirmahdifard commented:
@bear101 fixed. Due to my other open pull requests, I couldn't just coppy paste all files to make sure i've updated everything and I had to manually add the strings hear, That's why I've forgot a string. And that's one of the reasons I ask you to merge my pull requests more quickly. This is now fixed and compiles properly, I just tested again after adding the forgotan string to confurm that anything else is not missing, and it compiles.

amirmahdifard commented:
@bear101 is it ok now and working?

amirmahdifard commented:
@bear101 ok, this is now finished. Please try it and see everything is 100% ok, e.g dialog view for sited users etc, and then we can merge.

bear101 commented:
I'm not in favour of this change. It is not possible to see what properties are related. There's too much information in the channel activity. Also the channel activity class now has 52 member variables for controls.

amirmahdifard commented:
@bear101 Changes made in this activity are as follows:
• 
Moved all audio codec-related controls from the fragments to the main ChannelActivity.
• 
Renamed each control to include its related codec at the beginning, like opusaudiochan, opusapplication, speexsr, speexvbrsr, etc.
• 
Created a new method named ensureControlsInitialized to connect XML controls in code, set spinner options, and assign their event listeners.
• 
Called this method at the beginning of the activity creation.
• 
Used a variable to make sure ensureControlsInitialized runs only once.
• 
Edited the method exchangeChannel to get values from the controls and set them on the channel, and vice versa.
• 
Removed the "audio codec" button from the activity.
• 
Added a variable named audioCodec to directly access channel.audioCodec (just for convenience).
• 
Added an event listener to the codec type spinner. When a codec is selected, only the related controls will be enabled and shown, while the others will be hidden or disabled using setEnabled and setVisibility inside onItemSelected method of OnItemSelectedListener.
• 
Added another event to the fixvol checkbox to show or hide the fixvolume slider based on its state using the same setEnabled and setVisibility methods.
• 
Checked the state of the "Join channel on exit" checkbox in onOptionsItemSelected and acted accordingly:
• 
If checked, used doJoin() to create and join the channel.
• 
If not, used doMakeChannel() to only create the channel.Note: The only thing that might be visually misconfigured is the layout in the XML, because we are blind and just copied elements from fragment XML files into the activity XML.
Now, please talk about what needs changing, we'll do it.

amirmahdifard commented:
@bear101 ok I understand you, but, first, the user is not gonna see all of the controles, for example when no audio is selected, all audio codec related are being disabled and completely disappeared from the screen. Second, you like the clients to be match each other, and channel dialog in windows, both teamtalk classic and qt teamtalk is exactly like this, they both have the same amount of controles, so with this pull request, channel dialog will look the same in both android and desktop so what's the problem with it?  3th, all other similar dialogs, both in android and desktop clients, are desighned the same. for example, server entry dialog on android, channel dialog on windows, and server prop dialog in windows. 4th, i've wrote this activity due to so many users that are using teamtalk android dayly, and when I gave them an apk test file so they can try it, and they all liked. In my opinion teamtalk is open source and developers are working on it freely, I think we should do what moste users of the client wants. And this activity no longer has the bugs of the old activity, it's working bug free, and clean, and it contanes all of the new options, such as stream timeout timers. What about now? would you acsept this please given the above provided reasons?

amirmahdifard commented:
the thing that I don't understand is, that the channel dialog in qt client exactly has the same number of controles as this activity, so why do you have problem with this? it's a bit strange.

beqabeqa473 commented:
Mobile clients should be ergonomic by design unlike desktop one, in which there can be larger screen and possibility to quickly move from one pane to another.

You can just make tabbed activity or nivent something to be more ergonomic and easy to use.

amirmahdifard commented:
if this pull request is not gonna be acsepted, it's disrespect to my time and coding. Before I start to implement this, I asked question about it, and people agreed to this to be like windows, use combo boxes, etc. Then I took my time and implemented this as clean as possible. Even people requested the developer of teamtalk pro to make the channel prop activity to be like this because they were complaning that channel prop activity of teamtalk android is buggy so much. you aren't acsepting changes, so people are being forced to creat other clients, like teamtalk pro.

Nardol commented:
Why not making tab as suggested before? I ask only to understand, there is not critisisme from me about your work.

amirmahdifard commented:
@Nardol because currently it's tabbed, there is no change needed by me. closing this pull request makes it tabbed, and all my work is lost

beqabeqa473 commented:
You can just fix bugs and add missing functionality. No need to make ui fat.

Nardol commented:
> @Nardol Saying that it should be a TabPage is not acceptable.

@mohammadhoseinGhezelsofla (because it looks tagging people without real need is your way to do) Mentioning me to say I have told something I did not is not fair nor acceptable.
I asked a question and did not give any opinion. I only wished to understand after receiving some GitHub notification by mail about the same subject.
Recently I decided to unblock @amirmahdifard on GitHub hoping some change but maybe I should revert and block again to avoid this kind of situation and being falsely quoted by a member of his fanclub because I tried to understand.

amirmahdifard commented:
@mohammadhoseinGhezelsofla exactly i'm saying the same, this activity cannot be tabbed because making such an activity tabbed looks very strange, plus it's the same on windows, and we desighned it the same as windows, and in my opinion, this activity is correctly working, and the maintaner of the project haven't red my code at all, and just being disagreeing with it, with the reason of 52 controles. in qt client, and classic client, we exactly have the same amount of controles but they become visible and enabled, and invisible and disabled in different conditions what's been selected in audio codec combo box. So, what you are saying about 52 controles is not acseptable at all.

jeremyp3 commented:
And now the threats are coming...

amirmahdifard commented:
@hwangsihu hi bro. As I know, you also have android studio and you can build teamtalk android. Can you please, clone this branch, build it and install it on your android device, and go to the channel prop dialog, and see if you can find a single bug over there? as a teamtalk user, try to creat or update a channel, and see is it bad or good, it looks exactly like the desktop clients. Thanks!

CoBC commented:
As far as I can see, nobody has exposed any bug in this PR, just point the fact that the view now contains a lot of controls and the GUI becomes less understandable for users.
If you don't accept constructive comments, maybe you should reconsidere your place to contribute on open source projects...

Muamalaljanahi commented:
@amirmahdifard Your PR was not merged for a clear reason Sending messages repeatedly will not change that.
This is disruptive to those who get notifications for this repository and you will look like a nuisance, Maybe you should review your behavior because you are the only one who sends messages like this and tries to force everyone to do what you want even if it annoys others
I have'nt seen similar behavior on this repository or on any other GitHub repository I follow
Many people have warned you about that several times hoping you would stop but it was to no avail.

amirmahdifard commented:
hello corentin, and you are correct, and I've answered why it has more controles, all controles won't be shown to the user at once, like qt client, and classic client. channel dialog in android now looks the same as qt and classic client.

hwangsihu commented:
> @hwangsihu hi bro. As I know, you also have android studio and you can build teamtalk android. Can you please, clone this branch, build it and install it on your android device, and go to the channel prop dialog, and see if you can find a single bug over there? as a teamtalk user, try to creat or update a channel, and see is it bad or good, it looks exactly like the desktop clients. Thanks!

Oh sorry sorry, Why didn't I receive the email? I just checked, so there's no need to test it now, right?

hwangsihu commented:
@amirmahdifard, Can you give me permission to access that repository? I want to fix the error, but I don't think I can because I don't have write permission for the organization.

amirmahdifard commented:
@hwangsihu when I compile, it doesn't give any error, but ok, I checked the box so that you can push to this branch.

hwangsihu commented:
@amirmahdifard
Thank you, but I don't have maintainer permissions for this repository, so you will need to grant me permissions directly for your repository, or @bear101 will need to grant me permissions for that organization (but that may be too difficult a decision, so the former would be much better).

diamondStar35 commented:
Greetings everyone.
I firstly want to understand why so many people complain about having too much controls in the UI, when in fact, not everything is shown to the user. This is simply drama and the developer doesn't even care.
@bear101 Your complains for closing this pull request are not reasonable at all, it's just kind of disregard to developers who work to improve your bad TeamTalk5 client which you don't even care about, at least for android. and when somebody else trys to help, you simply close their pull request without any reason. I don't understand what is this childish drama and complains.
Have you even run the code before complaining so much? Have you checked the author's comments? No, and i'm saying no because your complains have no kind of point. The fact that there are 52 controls in one activity is not a point, there are at maximum, 20 controls shown at the same time.
In the end, it's a waste of time for someone who's trying to work with you.

hwangsihu commented:
However, because of the translation string, I requested he fix it **by showing me the log**, but he claimed there was no error. Instead of fixing the error himself, he granted write permissions to his branch to people at BearWare.dk who had write permissions.
But that's not the problem. I'm also a company employee, but I don't have write permissions. So I requested he grant me permissions so I could fix the error myself, but he didn't listen, and time just passed.

Did @bear101 simply close the PR with the error?

I apologize if my understanding is incorrect.

hwangsihu commented:
HMMM...
